addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/XxeScanRule.java

```diff
@@ -272,6 +282,18 @@ private void outOfBandFileInclusionAttack() {
                 payload = MessageFormat.format(ATTACK_MESSAGE, "https://" + oastPayload);
                 msg.setRequestBody(payload);
                 sendAndReceive(msg);
+
+                // Try again with parameter entity and http
+                msg = getNewMsg();
+                payload = MessageFormat.format(PARAMETER_ENTITY_ATTACK_MESSAGE, "http://" + oastPayload);
+                msg.setRequestBody(payload);
+                sendAndReceive(msg);
```

Comment:
I think it might be better to use a new alert/payload, otherwise the message will have a misleading request body. If not, a method should be extracted to avoid duplicating this logic 3 times.